admin/autoupdater/src/util.c

```diff
@@ -55,13 +58,20 @@ void run_dir(const char *dir) {
 			dup2(null_fd, 1);
 			dup2(null_fd, 2);
 			close(null_fd);
-			execl(*path, *path, (char *)NULL);
-			fprintf(stderr, "autoupdater: warning: failed executing %s: ", *path);
-			perror(NULL);
+			execl(globbuf.gl_pathv[i], globbuf.gl_pathv[i], (char *)NULL);
 			exit(EXIT_FAILURE);
-		} else if (waitpid(pid, NULL, 0) != pid) {
-			fprintf(stderr, "autoupdater: warning: failed waiting for child %d: ", pid);
-			perror(NULL);
+		} else {
+			int wstatus;
+			if (waitpid(pid, &wstatus, 0) != pid) {
+				fprintf(stderr, "autoupdater: warning: failed waiting for child %d corresponding to %s: ", pid, globbuf.gl_pathv[i]);
+				perror(NULL);
+			}
+			else if (!WIFEXITED(wstatus)) {
+				fprintf(stderr, "autoupdater: warning: execution of %s exited abnormally\n", globbuf.gl_pathv[i]);
+			}
+			else if (WEXITSTATUS(wstatus)) {
+				fprintf(stderr, "autoupdater: warning: execution of %s exited with status code %d\n", WEXITSTATUS(wstatus));
```

Comment:
Missing `globbuf.gl_pathv[i]` argument.

I guess it would make sense to assign the path to a variable at the top of the loop body instead of writing `globbuf.gl_pathv[i]` each time.